packages/gatsby-theme-docs/src/components/card.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from '@emotion/styled';
+import css from '@emotion/css';
+import { navigate } from '@reach/router';
+import SpacingsStack from '@commercetools-uikit/spacings-stack';
+import SpacingsInline from '@commercetools-uikit/spacings-inline';
+import { designSystem } from '@commercetools-docs/ui-kit';
+import Link from './link';
+
+const flatStyle = css`
+  border: 1px solid ${designSystem.colors.light.borderSecondary};
+`;
+const clickableStyle = css`
+  cursor: pointer;
+  box-shadow: ${designSystem.tokens.shadowForClickableCard};
+`;
+const CardContainer = styled.div`
+  ${(props) => (props.clickable ? clickableStyle : flatStyle)};
+  background-color: ${designSystem.colors.light.surfacePrimary};
+  padding: ${designSystem.dimensions.spacings.m};
+  border-radius: ${designSystem.tokens.borderRadiusForCard};
+`;
+const Icon = styled.div`
+  width: ${designSystem.dimensions.spacings.big};
+  min-width: ${designSystem.dimensions.spacings.big};
+  height: ${designSystem.dimensions.spacings.big};
+
+  svg {
+    width: 100%;
+    height: 100%;
+  }
+`;
+const normalTitle = css`
+  font-size: ${designSystem.typography.fontSizes.h3};
+  line-height: ${designSystem.typography.lineHeights.cardNormalTitle};
+`;
+const smallTitle = css`
+  font-size: ${designSystem.typography.fontSizes.h4};
+  line-height: ${designSystem.typography.lineHeights.cardSmallTitle};
+`;
+const Title = styled.h6`
+  ${(props) => (props.smallTitle ? smallTitle : normalTitle)};
+  font-weight: ${designSystem.typography.fontWeights.medium};
+  letter-spacing: 0;
+`;
+const ReadMore = styled.div`
+  border-top: 1px solid ${designSystem.colors.light.borderSecondary};
+  padding-top: ${designSystem.dimensions.spacings.m};
+`;
+
+const Card = (props) => {
+  return (
+    <CardContainer
+      {...props[0]}
+      onClick={props[0].clickable ? handleClick : undefined}
+    >
+      {props.icon ? renderCardContentWithIconLayout() : renderCardContent()}
+    </CardContainer>
+  );
+
+  function handleClick(e) {
+    e.preventDefault();
+    navigate(props.href);
```

Comment:
Why not wrapping the card with a `<Link>` component or something? It would also be semantically more correct.

Reply:
Yes, it should be a real link but It could be that we have to refactor the Link code a bit to separate the behaviour from the Styling.  

It's not ideal for accessibility to the have the whole card content be the link text (the screenreader then reads out the whole text twice) but if the card does not have a text link at the bottom it would go completely unnoticed as a link so wrapping the whole thing is better IMHO.

Reply:
There is an invalid DOM nesting problem of `<a>` of the text link being a child of the `<a>` from the `Link` component.

Reply:
PS:  There's a formal problem though:  https://html.spec.whatwg.org/#the-a-element  

>"there must be no interactive content descendant, a element descendant, or descendant with the tabindex attribute specified."

So strictly spoken, since the content of the card can contain links, we would only be on the safe side not making the card wrapped in a link and tagging the javascript link using aria spec attributes:

Would the following pseudo-html make sense? 

```html
<ul class="cards-wrapper">
   <li class="card" 
        role="link" 
        aria-label="The Card Title (if available)" 
        onClick="theSameCodeWeUseInOurLinkComponent('/foo/bar')">
           <h6>The Card Title</h5>
           <p>All the <strong>formatted</strong> markdown card body</p>
           <a href="/foo/bar">The textLink if passed</a>
    </li>
 </ul>
```

Reply:
Well the solution is easy: wrap the card with a link if there is no "bottom" link, otherwise the card is a div

Reply:
I guess I can just copy the link style instead and apply it to a different DOM element for the text link.

Reply:
Otherwise use aria attributes to indicate that the card is a link (if there is no "bottom" link).

Reply:
> Well the solution is easy: wrap the card with a link if there is no "bottom" link, otherwise the card is a div

The card content is markdown, that's not safe either.  We'll have to live with it being a "fake" link.  It's not a problem for crawlers, the pages we want indexed are in the nav anyways.

Reply:
What I have no idea how to solve cleanly is how to properly cover all the functional behavior the `PureLink` has (  https://github.com/commercetools/commercetools-docs-kit/blob/master/packages/gatsby-theme-docs/src/components/link.js#L97 ) 

If we don't want to duplicate it all here's an idea ( I have no idea whether this works in react): We could _always_ instantiate the textLink  at the bottom if an `href` is present.  `textLink` would then only trigger whether it's visible or not.   A click event on the card would be passed through to the text link,  which would make sure than any necessary behavior (including google analytics,  behavior buried in the GatsbyLink that it's using etc.) would be used and nothing is duplicated.  

@emmenko  crazy or useful?

Reply:
> A click event on the card would be passed through to the text link,

PS: this was a "jQuery-ish" way to word it where you can randomly wire events inside the DOM without constraints.   My naive idea in React would be that the Card component acquires a ref to the virtual DOM element of the `<Link ...>` it has as a child and then calls the `click()` on that ref when the whole card is clicked.   But I bet there's more proper ways to handle it.

Reply:
It seems overly complicated to me. Why not “the card is a link or a div”?
Yes, the content might include links but is that even a use case we want to support?

Reply:
Cards that are clickable and have a text link at the bottom that points to the same location are a known and intended use case because there's no hover effect on touch devices so for the primary call to action you want to have the link (and the text that makes clearer where you navigate to). 

But disallowing links in the card body content would be OK though (a.k.a. "it just fails", I would not even want to build the logic to catch it)

Same link on both the card and the bottom could be done by the "inverted" hack that the card is a link and the link itself isn't a link but just looks like one.  Not sure if that is nicer implementation wise, but it's clearly clean from the DOM and HTML perspective! 

So for illustration does this look acceptable as pseudo-html:

**Clickable:**
```html
<li class=card>
  <a href="/foo/bar" class="LinkComponent">
     <h6>Title</h6>
     <p>markdown <strong>formatted</strong> content but no links possible. 
     <span class="fake-link-that-changes-color-on-hover">Read more about cards...</span>
  </a>
</li>
```

**Not clickable:**
```html
<li class=card>
     <h6>Title</h6>
     <p>markdown <strong>formatted</strong> content and <a href="/baz" class="LinkComponent">links</a> possible. 
     <a href="/foo/bar" class="LinkComponent">Read more about cards...</a>
</li>
```

Reply:
Yes, the pseudo-html output is what I would expect as well 👌